src/main/scala/utility/document/html/HTML.scala

```diff
@@ -60,6 +63,18 @@ class HTMLDom private (htmlDocument: org.jsoup.nodes.Document):
   def getElementByClass(className: String): Seq[HTMLElement] =
     htmlDocument.getElementsByClass(className).asScala.map(HTMLElement(_)).toSeq
 
+  /**
+   * Retrieves all occurrences of URLs in the HTML document.
+   *
+   * @return a sequence of URLs representing the occurrences found.
+   */
+  def getAllLinkOccurrences(): Seq[URL] = 
+    val hrefLinks = htmlDocument.select("[href]").eachAttr("href").asScala.toSeq.map(_.toUrl)
+    val srcLinks = htmlDocument.select("[src]").eachAttr("src").asScala.toSeq.map(_.toUrl)
+    val urlPattern: Regex = new Regex("(https?://\\S+|http://\\S+|www\\.\\S+)")
```

Comment:
Use a more simple way to express regex: 
[Expression Patterns](https://docs.scala-lang.org/tour/regular-expression-patterns.html)
you must use "(https?://\\S+|http://\\S+|www\\.\\S+)".r